nixos/lib/test-driver/doc.py

```diff
@@ -0,0 +1,63 @@
+import ast
+import sys
+
+"""
+This program takes all the Machine class methods and prints its methods in
+markdown-style. These can then be included in the NixOS test driver
+documentation, which is then always in sync with the code.
```

Comment:
```suggestion
markdown style, assuming the docstrings themselves are also in markdown.

These are included in the test driver documentation in the NixOS manual.
See https://nixos.org/manual/nixos/stable/#ssec-machine-objects
```

I think synchronicity is implied.

Reply:
I haven't thoroughly reviewed the python code below, but it looks like it gets the job done.
A bit ad hoc perhaps, but that's fine for the currently limited purpose.

Reply:
It's ad hoc, yes. The existing tools for docstring-to-markdown generation partly didn't like that we're using new things (e.g. the `|` sum type combinator) and also generated too much which would need to be filtered. so i figured that this approach here is even simpler and very easy to customize.

Reply:
Oh, i get it... you mean in the event of more stuff being extracted, having one deriv with multiple output files in the same output folder. yes.